sqlite_helpers.py

```diff
@@ -19,28 +22,53 @@ def create_table():
 
 def insert_url(url: str, alias: str):
     time = datetime.datetime.now()
+    conn = create_connection()
+    c = conn.cursor()
     with conn:
         c.execute("INSERT INTO urls (url, alias, timestamp) VALUES (?, ?, ?)", (
                   url, alias, time))
 
 
 def delete_url(url: str):
+    conn = create_connection()
+    c = conn.cursor()
     with conn:
         c.execute("DELETE from urls WHERE url=?", (url,))
 
 
+def delete_alias(alias: str):
+    conn = create_connection()
+    c = conn.cursor()
+    with conn:
+        c.execute("DELETE from urls WHERE alias=?", (alias,))
+
+
 def list_urls():
-    c.execute("SELECT url from urls")
-    print(c.fetchall())
+    conn = create_connection()
+    c = conn.cursor()
+    with conn:
+        c.execute("SELECT url from urls")
+    return (c.fetchall())
```

Comment:
here we will want to return all values from the row since we might want to use it all later, not just the url. we will turn each row into a dictionary so we can get all the values easier later on.
turns `[(1, 'https://www.google.com/', 'Google', '2024-01-09 11:03:06.040248'), (2, 'https://github.com/', 'GitHub', '2024-01-09 11:03:06.042245')]` into `[{'id': 1, 'url': 'https://www.google.com/', 'alias': 'Google', 'created_at': '2024-01-09 11:03:06.040248'}, {'id': 2, 'url': 'https://github.com/', 'alias': 'GitHub', 'created_at': '2024-01-09 11:03:06.042245'}, {'id': 3, 'url': 'https://www.google.com/', 'alias': 'Google', 'created_at': '2024-01-09 11:04:01.926822'}, {'id': 4, 'url': 'https://github.com/', 'alias': 'GitHub', 'created_at': '2024-01-09 11:04:01.929824'}]`
```suggestion
    c = conn.cursor()
    c.execute("SELECT * from urls")
    result = c.fetchall()
    db_array = []
    for row in result:
      data = {
        "id": row[0]
        "url": row[1],
        "alias": row[2],
        "created_at": row[3]
      }
      db_array.append(data)
```